src/file_dialog.rs

```diff
@@ -518,8 +523,8 @@ impl FileDialog {
     ///
     /// Since `fs::canonicalize` is used, both absolute paths and relative paths are allowed.
     /// See `FileDialog::canonicalize_paths` for more information.
-    pub fn initial_directory(mut self, directory: PathBuf) -> Self {
-        self.config.initial_directory.clone_from(&directory);
+    pub fn initial_directory(mut self, directory: &PathBuf) -> Self {
+        self.config.initial_directory.clone_from(directory);
```

Comment:
The method parameter should not be changed as it is a breaking change.

I would instead remove the `clone_from`:
```suggestion
    pub fn initial_directory(mut self, directory: PathBuf) -> Self {
        self.config.initial_directory = directory;
```